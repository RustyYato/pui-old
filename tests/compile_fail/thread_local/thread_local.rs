pui::make_typeid_tl! {
    type Id;
}

fn main() {
    fn assert_send_sync<T: Send + Sync>(t: T) {}
    assert_send_sync(Id::new());
}
