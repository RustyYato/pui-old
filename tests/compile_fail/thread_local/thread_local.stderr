error[E0277]: `*mut ()` cannot be sent between threads safely
 --> $DIR/thread_local.rs:7:5
  |
6 |     fn assert_send_sync<T: Send + Sync>(t: T) {}
  |                            ---- required by this bound in `assert_send_sync`
7 |     assert_send_sync(Id::new());
  |     ^^^^^^^^^^^^^^^^ `*mut ()` cannot be sent between threads safely
  |
  = help: within `pui::typeid_tl::Type<Id>`, the trait `Send` is not implemented for `*mut ()`
  = note: required because it appears within the type `ThreadLocal`
  = note: required because it appears within the type `(pui::Invariant<Id>, ThreadLocal)`
  = note: required because it appears within the type `PhantomData<(pui::Invariant<Id>, ThreadLocal)>`
  = note: required because it appears within the type `pui::typeid_tl::TypeHandle<Id>`
  = note: required because it appears within the type `pui::typeid_tl::Type<Id>`

error[E0277]: `*mut ()` cannot be shared between threads safely
 --> $DIR/thread_local.rs:7:5
  |
6 |     fn assert_send_sync<T: Send + Sync>(t: T) {}
  |                                   ---- required by this bound in `assert_send_sync`
7 |     assert_send_sync(Id::new());
  |     ^^^^^^^^^^^^^^^^ `*mut ()` cannot be shared between threads safely
  |
  = help: within `pui::typeid_tl::Type<Id>`, the trait `Sync` is not implemented for `*mut ()`
  = note: required because it appears within the type `ThreadLocal`
  = note: required because it appears within the type `(pui::Invariant<Id>, ThreadLocal)`
  = note: required because it appears within the type `PhantomData<(pui::Invariant<Id>, ThreadLocal)>`
  = note: required because it appears within the type `pui::typeid_tl::TypeHandle<Id>`
  = note: required because it appears within the type `pui::typeid_tl::Type<Id>`
